nodes/modifier_make/subdivide_mk2.py

```diff
@@ -0,0 +1,399 @@
+
+        onew = self.outputs.new
+        onew('SvVerticesSocket', 'Vertices')
+        onew('SvStringsSocket', 'Edges')
+        onew('SvStringsSocket', 'Faces')
+        onew('SvStringsSocket', 'FaceData')
+
+        onew('SvVerticesSocket', 'NewVertices')
+        onew('SvStringsSocket', 'NewEdges')
+        onew('SvStringsSocket', 'NewFaces')
+
+        onew('SvVerticesSocket', 'OldVertices')
+        onew('SvStringsSocket', 'OldEdges')
+        onew('SvStringsSocket', 'OldFaces')
+
+        self.update_mode(context)
+
+    def mute_node(self):
+        origin_socket = [0, 1, 2, 3, 0, 1, 2, 0, 1, 2]
+
+        for i, s in zip(origin_socket, self.outputs):
+            if self.inputs[i].is_linked:
+                s.sv_set(self.inputs[i].sv_get(deepcopy=False))
+
+    def freeze_node(self):
+        if 'storage' in self.mem.keys():
+            for m, s in zip(self.mem['storage'], self.outputs):
+                s.sv_set(m)
+            return True
+        return False
+
+    def get_result_pydata(self, geom):
+        new_verts = [v for v in geom if isinstance(v, BMVert)]
+        new_edges = [e for e in geom if isinstance(e, BMEdge)]
+        new_faces = [f for f in geom if isinstance(f, BMFace)]
+
+        new_verts = [tuple(v.co) for v in new_verts]
+        new_edges = [[edge.verts[0].index, edge.verts[1].index] for edge in new_edges]
+        new_faces = [[v.index for v in face.verts] for face in new_faces]
+
+        return new_verts, new_edges, new_faces
+
+    def get_data(self):
+        inputs = self.inputs
+
+        verts_s = inputs['Vertices'].sv_get(deepcopy=False)
+        edges_s = inputs['Edges'].sv_get(default=[[]], deepcopy=False)
+        faces_s = inputs['Faces'].sv_get(default=[[]], deepcopy=False)
+
+        masks_s = inputs['EdgeMask'].sv_get(default=[[1]], deepcopy=True)
+        face_data_s = inputs['FaceData'].sv_get(default=[[]], deepcopy=True)
+
+        cuts_s = inputs['Cuts'].sv_get(deepcopy=False)[0]
+        smooth_s = inputs['Smooth'].sv_get(deepcopy=False)[0]
+        fractal_s = inputs['Fractal'].sv_get(deepcopy=False)[0]
+        along_normal_s = inputs['AlongNormal'].sv_get(deepcopy=False)[0]
+        seed_s = inputs['Seed'].sv_get(deepcopy=False)[0]
+
+        return match_long_repeat([verts_s, edges_s, faces_s, face_data_s, masks_s, cuts_s, smooth_s, fractal_s, along_normal_s, seed_s])
+
+    def dont_process(self):
+
+        inputs, outputs = self.inputs, self.outputs
+        if not (any(s.is_linked for s in outputs) and all(s.is_linked for s in inputs[:2])):
+            return True
+        if self.sv_mute:
+            self.mute_node()
+            return True
+        if self.sv_frezze and self.sv_record:
+            if self.freeze_node():
+                return True
+        elif self.sv_frezze:
+            self.sv_record = True
+            return False
+
+    def process(self):
+        inputs, outputs = self.inputs, self.outputs
+
+        if self.dont_process():
+            return
```

Comment:
This is no great method. When you are calling like this `if sekf.dont_process()` , I expect that the purpose of this method is return bool value. But actually it is doing some extra job inside. I recommend to split this method on two or more. `dont_process` method should only checking whether it should return `True` or `False` nothing more.

You have lost 2 `else` conditions there.

Reply:
you are right is not the most elegant... finally i removed the properties so less hassle.
> You have lost 2 else conditions there.

[No else after return](https://pheanex.github.io/pylint/#R1705)